src/daft-connect/src/functions/core.rs

```diff
@@ -0,0 +1,105 @@
+        parent.add_fn("<=", BinaryOpFunction(Operator::LtEq));
+        parent.add_fn(">", BinaryOpFunction(Operator::Gt));
+        parent.add_fn(">=", BinaryOpFunction(Operator::GtEq));
+        parent.add_fn("+", BinaryOpFunction(Operator::Plus));
+        parent.add_fn("-", BinaryOpFunction(Operator::Minus));
+        parent.add_fn("*", BinaryOpFunction(Operator::Multiply));
+        parent.add_fn("/", BinaryOpFunction(Operator::TrueDivide));
+        parent.add_fn("//", BinaryOpFunction(Operator::FloorDivide));
+        parent.add_fn("%", BinaryOpFunction(Operator::Modulus));
+        parent.add_fn("&", BinaryOpFunction(Operator::And));
+        parent.add_fn("|", BinaryOpFunction(Operator::Or));
+        parent.add_fn("^", BinaryOpFunction(Operator::Xor));
+        parent.add_fn("<<", BinaryOpFunction(Operator::ShiftLeft));
+        parent.add_fn(">>", BinaryOpFunction(Operator::ShiftRight));
+        parent.add_fn("isnotnull", UnaryFunction(|arg| arg.not_null()));
+        parent.add_fn("isnull", UnaryFunction(|arg| arg.is_null()));
+        parent.add_fn("not", UnaryFunction(|arg| arg.not()));
+        parent.add_fn("sum", UnaryFunction(|arg| arg.sum()));
+        parent.add_fn("mean", UnaryFunction(|arg| arg.mean()));
+        parent.add_fn("stddev", UnaryFunction(|arg| arg.stddev()));
+        parent.add_fn("min", UnaryFunction(|arg| arg.min()));
+        parent.add_fn("max", UnaryFunction(|arg| arg.max()));
+        parent.add_fn("count", CountFunction);
+    }
+}
+
+pub struct BinaryOpFunction(Operator);
+pub struct UnaryFunction(fn(ExprRef) -> ExprRef);
+pub struct CountFunction;
+
+impl SparkFunction for BinaryOpFunction {
+    fn to_expr(
+        &self,
+        args: &[Expression],
+        analyzer: &SparkAnalyzer,
+    ) -> eyre::Result<daft_dsl::ExprRef> {
+        let args = args
+            .iter()
+            .map(|arg| analyzer.to_daft_expr(arg))
+            .collect::<eyre::Result<Vec<_>>>()?;
+
+        let [lhs, rhs] = args
+            .try_into()
+            .map_err(|args| eyre::eyre!("requires exactly two arguments; got {:?}", args))?;
+
+        Ok(binary_op(self.0, lhs, rhs))
+    }
+}
+
+impl SparkFunction for UnaryFunction {
+    fn to_expr(
+        &self,
+        args: &[Expression],
+        analyzer: &SparkAnalyzer,
+    ) -> eyre::Result<daft_dsl::ExprRef> {
+        match args {
+            [arg] => {
+                let arg = analyzer.to_daft_expr(arg)?;
+                Ok(self.0(arg))
+            }
+            _ => invalid_argument_err!("requires exactly one argument")?,
+        }
+    }
+}
+
+impl SparkFunction for CountFunction {
+    fn to_expr(
+        &self,
+        args: &[Expression],
+        analyzer: &SparkAnalyzer,
+    ) -> eyre::Result<daft_dsl::ExprRef> {
+        match args {
+            [arg] => {
+                let arg = analyzer.to_daft_expr(arg)?;
+
+                let arg = if arg.as_literal().and_then(|lit| lit.as_i32()) == Some(1i32) {
```

Comment:
Isn't `count(2)` also equivalent to `count(1)`, which is equivalent to `count(*)`? In fact, isn't `count(n)`, for any integer `n`, equivalent to `count(*)`?

Reply:
in spark, it is only ever sent over as `count(1)`, and it never allows any args

https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.count.html

so it can only ever be called via
```py
df.count()
```

which gets serialized via protobuf as `count(1)`. So this is really about special casing for what spark connect is sending us vs how we internally represent a `count(*)` / `count(n)`